platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/platform/DefaultDoubleAccumulator.java

```diff
@@ -20,39 +20,44 @@
 import static org.apache.commons.lang3.builder.ToStringStyle.SHORT_PREFIX_STYLE;
 
 import com.swirlds.common.metrics.DoubleAccumulator;
+import com.swirlds.common.metrics.atomic.AtomicDouble;
 import com.swirlds.common.metrics.platform.Snapshot.SnapshotEntry;
 import java.util.List;
+import java.util.function.DoubleBinaryOperator;
+import java.util.function.DoubleSupplier;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 /**
  * Platform-implementation of {@link DoubleAccumulator}
  */
 public class DefaultDoubleAccumulator extends DefaultMetric implements DoubleAccumulator {
 
-    private final java.util.concurrent.atomic.DoubleAccumulator container;
-    private final double initialValue;
+    private final AtomicDouble container;
+    private final DoubleBinaryOperator accumulator;
+    private final DoubleSupplier initializer;
 
     public DefaultDoubleAccumulator(final Config config) {
         super(config);
-        this.container =
-                new java.util.concurrent.atomic.DoubleAccumulator(config.getAccumulator(), config.getInitialValue());
-        this.initialValue = config.getInitialValue();
+        this.accumulator = config.getAccumulator();
+        final double initialValue = config.getInitialValue();
+        this.initializer = config.getInitializer() != null ? config.getInitializer() : () -> initialValue;
```

Comment:
Since config is not a record 2 calls to `config.getInitializer()` could result in 2 different values. in worst case the second call could return `null`. Therefore I would refactor this to only have 1 call to `config.getInitializer()`